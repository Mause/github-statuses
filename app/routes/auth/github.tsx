// app/routes/auth/github.tsx
import type { ActionFunction} from "@remix-run/node";
import { redirect } from "@remix-run/node";
import { authenticator } from "~/services/auth.server";

export async function loader() {
  return redirect("/login");
}

export const action: ActionFunction = async ({ request }) => {
  return authenticator().authenticate("github", request);
};
