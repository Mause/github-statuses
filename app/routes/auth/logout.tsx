import { Button } from "@primer/react";
import type { ActionArgs } from "@remix-run/node";
import { Form } from "@remix-run/react";
import { Wrapper } from "~/components";
import { authenticator } from "~/services/auth.server";

export const action = async ({ request }: ActionArgs) =>
  authenticator().logout(request, {
    redirectTo: "/login",
  });

export default function Logout() {
  return (
    <Wrapper>
      {<></>}
      <Form method="post">
        <Button type="submit">Logout with GitHub</Button>
      </Form>
    </Wrapper>
  );
}
