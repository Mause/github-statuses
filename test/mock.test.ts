import { Octokit } from "@octokit/rest";
import { request } from "./auth.test";
import { vi } from "vitest";

it("request", async () => {
  const resp = await request("https://api.github.com/user");
  expect(resp).toMatchSnapshot();
});
it("request with octokit", async () => {
  globalThis.fetch = vi.fn().mockImplementation(async (url) => {
    const res = await request(url);

    return {
      ...res,
      headers: new Headers(res.headers),
      text: function () {
        return JSON.stringify(this.data);
      },
    };
  });
  const octokit = new Octokit();
  const resp = await octokit.request("https://api.github.com/user");
  delete resp.headers.date;
  expect(resp).toMatchSnapshot();
});
