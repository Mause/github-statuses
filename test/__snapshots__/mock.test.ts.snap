// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`auth > make login request 1`] = `
[
  [
    "scope",
    "scope",
  ],
  [
    "allow_signup",
    "true",
  ],
  [
    "response_type",
    "code",
  ],
  [
    "client_id",
    "clientId",
  ],
  [
    "redirect_uri",
    "http://localhost:3000/callback",
  ],
]
`;

exports[`auth > make login request 2`] = `"https://github.com/login/oauth/authorize?scope=scope&allow_signup=true&response_type=code&client_id=clientId&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fcallback"`;

exports[`auth > should be able to call strategy.authenticate 1`] = `
{
  "login": "login",
}
`;

exports[`request 1`] = `
{
  "data": {
    "id": 690,
    "login": "Mause",
    "name": "Elli",
  },
  "headers": {
    "date": "Fri, 19 Jan 2024 07:12:21 GMT",
  },
  "status": 200,
  "url": "https://api.github.com/user",
}
`;

exports[`request with octokit 1`] = `
{
  "data": "{"id":690,"login":"Mause","name":"Elli"}",
  "headers": {},
  "status": 200,
  "url": "https://api.github.com/user",
}
`;
